lightning/src/ln/channelmanager.rs

```diff
@@ -2294,6 +2440,9 @@ impl<Signer: Sign, M: Deref, T: Deref, K: Deref, F: Deref, L: Deref> ChannelMana
 				if let Some((err, code, chan_update)) = loop {
 					let forwarding_id = match id_option {
 						None => { // unknown_next_peer
+							if fake_scid::is_valid_phantom(self.fake_scid_offset, *short_channel_id) {
+								break None
```

Comment:
Hmmm, I feel like we should hit most of the checks below, even if we don't have a channel.  Two reasons - a) do we currently check that there was a non-negative fee/CLTV delta at the second-to-last (phantom) hop here? b) it's kinda nice to enforce pretending we're a real channel, no? Will take some refactoring of the code below, though.

Reply:
> a) do we currently check that there was a non-negative fee/CLTV delta at the second-to-last (phantom) hop here?

Hmm, so we don't do this check, but also I'm not sure how we would do this given `incorrect_cltv_expiry` onion error includes a ChannelUpdate for the forwardee channel?

Reply:
Hmm...what does the spec say about errors for private channels here? Its obviously kinda nonsense to give a channelupdate for a private channel...maybe we just need to update the error here to not return anything for private channels? May need a spec change, but I don't see why we shouldn't do it unilaterally for now.

Reply:
Discussed offline. Spec says "Please note that the channel_update field is mandatory in messages whose failure_code includes the UPDATE flag.", i.e. channel updates are currently required even for private channels. 

Plan is:
* push a spec change through to address this issue
* implement the proposed spec change in this PR